src/Data/MIME.hs

```diff
@@ -679,6 +680,31 @@ headerDate = lens getter setter
         . C8.unpack . view (header "date")
     setter hdrs x = set (header "date") (renderRFC5422Date x) hdrs
 
+-- | Returns a space delimited `B.ByteString` with values from identification
+-- fields from the parents message `Headers`. Rules to gather the values are in
+-- accordance to RFC5322 - 3.6.4 as follows sorted by priority (first has
+-- precedence):
+-- * Values from 'References' and `Message-ID` (if any)
+-- * Values from 'In-Reply-To' and 'Message-ID' (if any)
+-- * Value from 'Message-ID' (in case it's the first reply to a parent mail)
+-- * otherwise Nothing is returned indicating that the replying mail should not have a 'References' field.
+--
+replyHeaderReferences :: Getter Headers (Maybe C8.ByteString)
+replyHeaderReferences = to go
+  where
+    go :: Headers -> Maybe C8.ByteString
+    go hdrs =
+      ((\x y -> x <> " " <> y)
```

Comment:
Meh - feels shoe-horned :(. I've used `toListOf` first, but using `(<|>)` on lists will merge them if both are not empty. What am I missing I wonder ... any pointers what to library/Lens I should look into?

Reply:
From my reading of the spec, I'd express it like this:
```haskell
replyHeaderReferences = to $ \hdrs ->
  let
    xs = catMaybes
      [ preview (header "references") hdrs <|> preview (header "in-reply-to") hdrs
      , preview (header "message-id") hrds
      ]
  in
    if null xs then Nothing else Just (B.intercalate " " xs)
```

Note there is one requirement not addressed: you can only use the "in-reply-to"
header if it contains one message identifier.  I'm happy for us not to bother with
that check, but it should be commented in the code.

Reply:
AAh yes! thank you so much. That is so much better.